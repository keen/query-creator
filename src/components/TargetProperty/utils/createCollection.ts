export const createCollection = (properties: Record<string, string>) =>
  Object.keys(properties).map((key: string) => ({
    propertyPath: key,
    propertyType: properties[key],
  }));
