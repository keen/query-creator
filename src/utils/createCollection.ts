export const createCollection = (properties: Record<string, string>) =>
  Object.keys(properties).map((key: string) => ({
    path: key,
    type: properties[key],
  }));
