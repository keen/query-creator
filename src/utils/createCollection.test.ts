import { createCollection } from './createCollection';

test('creates collection', () => {
  const collectionSchema = { date: 'String', userId: 'String' };

  expect(createCollection(collectionSchema)).toMatchSnapshot();
});
