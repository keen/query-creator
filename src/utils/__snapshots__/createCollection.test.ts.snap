// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates collection 1`] = `
Array [
  Object {
    "path": "date",
    "type": "String",
  },
  Object {
    "path": "userId",
    "type": "String",
  },
]
`;
